### ISSUE TITLE
[sklearn#34641] DOC: Add table to Recommendation Summary section in Common Pitfalls

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#304]
**Mirrored from [scikit-learn#34641](https://github.com/scikit-learn/scikit-learn/issues/34641)**

Original title: DOC: Add table to Recommendation Summary section in Common Pitfalls
Original labels: Documentation, Needs Triage
Created: 2026-08-03T10:36:25Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the issue linked to the documentation

In the [controlling pitfalls](https://scikit-learn.org/stable/common_pitfalls.html) page, in section 12.3, there is a recommendation summary for controlling randomness. I feel that while the note is okay, it makes sense to keep something simple to improve readability for beginners who need an easy reference to look at.

### Suggest a potential alternative/fix

Add a table to the recommendation summary section, below the paragraph. This will make it really easy for people to have an idea about some scenarios that they may run into and what they should handle. What I have so far is:

| Goal | Estimator | CV Splitter|
| ----- | ---- | ---- |
|  Same results every fit/split call  | int  | int |
| Robust CV of a randomized estimator (fresh RNG per fold) | RandomState or None   | int    |
| Comparable fold-to-fold scores across separate runs   | (any)  | int |
| Reproducible across program executions  | fixed RandomState or int | int |

Let me know if this is a good idea or there are any changes/additions to the table!

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Addressed in upstream PR(s). See scikit-learn/scikit-learn PRs for #34641.
